### DISCUSSION TITLE
How to get scene depth?

### DISCUSSION THREAD

**Discussion by Psi25** [bnpr/Malt#419] [Q&A] [answered]
I saw that there's a Depth input in the Render Output node. This got me thinking if I can extract the scene's depth data, but I don't know how.

**Comment by pragma37** [comment, marked as answer]
The PrePass node has a Normal Depth output texture, where the fourth channel is the current layer depth.
![imagen](https://user-images.githubusercontent.com/36610999/196045610-caaeaf8a-73ba-4a17-9ffd-fa2205a8cc57.png)
That texture is read internally in the Mesh MainPass shaders and in Screen shaders, but it's not currently exposed to nodes.
You could add an extra custom input and plug it there so it's available to nodes, or use a custom node function that reads it.
